### PR TITLE
[LED-94] Filtering displayed transactions by the selected account

### PR DESCRIPTION
**Pull request by shellajt**
You can now use the account dropdown to filter the transactions. Added a dummy account to allow the user to view all transactions if they want. Added a new db function and controller call for this.

### CODE REVIEW

File: src/main/java/ledger/database/storage/SQL/ISQLDatabaseTransaction.java [hunk 1]
```diff
@@ -213,6 +213,50 @@ default void editTransaction(Transaction transaction) throws StorageException {
...
+                    ", " + TransactionTable.TRANS_ACCOUNT_ID +
+                    ", " + TransactionTable.TRANS_PAYEE_ID +
+                    " FROM " + TransactionTable.TABLE_NAME +
+                    " WHERE " + TransactionTable.TRANS_ACCOUNT_ID + "=?;");
+            stmt.setInt(1, account.getId());
+            ResultSet rs = stmt.executeQuery();
+
+            ArrayList<Transaction> transactionList = new ArrayList<>();
+
+            while (rs.next()) {
```
Comment (cjm721) [inline src/main/java/ledger/database/storage/SQL/ISQLDatabaseTransaction.java]: Have this in another method also. Might as well abstract out now.

File: src/main/java/ledger/user_interface/ui_controllers/TransactionTableView.java [hunk 2]
```diff
@@ -281,13 +275,20 @@ public void handle(KeyEvent t) {
...
-        task.startTask();
-        List<Transaction> allTransactions = task.waitForResult();
+        List<Transaction> transactions;
+        if (accountFilter == null) {
+            TaskWithReturn<List<Transaction>> task = DbController.INSTANCE.getAllTransactions();
+            task.startTask();
+            transactions = task.waitForResult();
+        } else {
+            TaskWithArgsReturn<Account, List<Transaction>> task = DbController.INSTANCE.getAllTransactionsForAccount(accountFilter);
+            task.startTask();
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/TransactionTableView.java]: Can pull out these two lines out of the 'if'

**Comment by xxshadowfang** [comment]
So whenever an account is selected and you try to add an account the program errors about threads.

Also whenever you select an account there isnt a way to clear the filter to see all transactions again.

**Comment by cjm721** [comment]
Did you do a fresh pull?

**Comment by xxshadowfang** [comment]
Yeah its the latest. An option to see all only shows up when there are no accounts in the db

**Comment by LadyMopar** [comment]
I'm getting weird stuff too. I can import the CSV file just fine into one account, but if I go to import the QXF into the other account I get a bunch of errors in the console saying "ERROR - DbContoller.batchInsertTransaction: Adding transaction failed 
Error while adding transaction: no such Type defined" ... It's like the QFX isn't working at all. I'm also getting the same error on creating accounts as Garrett (although it goes ahead and creates the account anyway)

**Comment by shellajt** [comment]
@xxshadowfang Think your issue should be fixed

@LadyMopar Cj and I are both unable to recreate that bug. Can you give us a line number or something?

**Pull request merged**